src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -84,35 +76,66 @@ public IEnumerable<string> GetApplicableContentTypes(string method, string path)
 
         public Uri GetEffectivePath(string commandSpecifiedPath)
         {
-            if (Uri.TryCreate(commandSpecifiedPath, UriKind.Absolute, out Uri absoluteUri))
+            return GetEffectivePath(BaseAddress, string.Join('/', PathSections.Reverse()), commandSpecifiedPath);
+        }
+
+        public static Uri GetEffectivePath(Uri baseAddress, string pathSections, string commandSpecifiedPath)
+        {
+            // If an absolute uri string was already specified, just return that.
+            if (Uri.IsWellFormedUriString(commandSpecifiedPath, UriKind.Absolute))
             {
-                return absoluteUri;
+                return new Uri(commandSpecifiedPath, UriKind.Absolute);
             }
+            // If it wasn't, and there also isn't a base address, throw an exception
+            else if (baseAddress == null)
+            {
+                throw new ArgumentNullException(nameof(baseAddress), string.Format(Resources.Strings.HttpState_Error_NoAbsoluteUriNoBaseAddress, nameof(commandSpecifiedPath), nameof(baseAddress)));
+            }
+
+            // Get a builder for the base address
+            UriBuilder builder = new UriBuilder(baseAddress);
+
+            // Get everything beyond the BaseAddress for the current location
+            string path = pathSections;
 
-            UriBuilder builder = new UriBuilder(BaseAddress);
-            string path = string.Join('/', PathSections.Reverse());
+            // Split that off into the path and the querystring parameters (if any)
```

Comment:
nit: space between query and string here and ln 106

Reply:
Changed in 82e9fa7.